src/main/java/com/gmail/nossr50/runnables/player/PlayerProfileLoadingTask.java

```diff
@@ -42,9 +46,13 @@ public void run() {
         }
 
         PlayerProfile profile = mcMMO.getDatabaseManager().loadPlayerProfile(player.getName(), player.getUniqueId(), true);
+
         // If successful, schedule the apply
         if (profile.isLoaded()) {
             new ApplySuccessfulProfile(new McMMOPlayer(player, profile)).runTask(mcMMO.p);
+            Bukkit.getScheduler().runTask(mcMMO.p, () -> {
```

Comment:
Event should be fired async instead of being passed back to main, if the listener needs to be on main, that is their concern